aten/src/ATen/native/quantized/cpu/ACLUtils.h

```diff
@@ -0,0 +1,257 @@
+    std::rotate(
+        acl_dynamic_quant_cache.begin(),
+        acl_dynamic_quant_cache.begin() + 1,
+        acl_dynamic_quant_cache.end());
+    return acl_dynamic_quant_cache[0];
+  }
+
+ private:
+  // A 2 element (per layer) cache. Given it's not intended to store more than 2
+  // elements, we do not need a fancy implementation. The idea behind it is to
+  // allow for a (configuration free) fast path for autoregressive
+  // transformer-like models which usually involve 2 input tensor shapes; one
+  // for the prefill phase and another for the autoregressive phase
+  std::array<std::shared_ptr<ACLDynamicQuantMatmul>, 2> acl_dynamic_quant_cache;
+
+  std::shared_ptr<ACLDynamicQuantMatmul> create_acl_dynamic_quant_matmul(
+      const ACLDynamicQuantMatmulCacheKey& key) {
+    int64_t m =
+        std::get<static_cast<int>(ACLDynamicQuantMatmulCacheKeyIndex::M)>(key);
+    bool fuse_relu = std::get<static_cast<int>(
+        ACLDynamicQuantMatmulCacheKeyIndex::FUSE_RELU)>(key);
+    auto acl_gemm = std::make_shared<ACLDynamicQuantMatmul>();
+    acl_gemm->key = key;
+    acl_gemm->src_fp32_tensor_info = arm_compute::TensorInfo(
+        arm_compute::TensorShape(k_, m), arm_compute::Format::F32);
+
+    acl_gemm->src_fp32_tensor_info.set_are_values_constant(false);
+
+    acl_gemm->src_s8_tensor_info = arm_compute::TensorInfo(
+        arm_compute::TensorShape(k_, m),
+        1,
+        arm_compute::DataType::QASYMM8_SIGNED,
+        // TODO: setting the initial offset value to int8_t max instead of zero,
+        // because ACL currently skips MatrixBReduction calculation if the
+        // source offset at configuration time is zero. This is fixed by this
+        // PR: https://review.mlplatform.org/c/ml/ComputeLibrary/+/12820/8 This
+        // will be set to the actual src offset value at runtime.
+        arm_compute::QuantizationInfo(
+            1.0, std::numeric_limits<int8_t>::max(), true));
+    acl_gemm->src_s8_tensor_info.set_are_values_constant(false);
+
+    acl_gemm->wei_tensor_info = arm_compute::TensorInfo(
+        arm_compute::TensorShape(n_, k_),
+        1,
+        arm_compute::DataType::QASYMM8_SIGNED,
+        arm_compute::QuantizationInfo(wei_scale_, wei_zero_point_, true));
+    acl_gemm->wei_tensor_info.set_are_values_constant(true);
+
+    // True iff the linear layer has bias && all std::optional bias containers
+    // have a value
+    bool with_bias{false};
+    if (bias_.has_value()) {
+      acl_gemm->bia_tensor_info = arm_compute::TensorInfo(
+          arm_compute::TensorShape(1, n_), 1, arm_compute::DataType::F32);
+      acl_gemm->bia_tensor = arm_compute::Tensor();
+      with_bias = true;
+    }
+
+    acl_gemm->dst_tensor_info = arm_compute::TensorInfo(
+        arm_compute::TensorShape(n_, m), arm_compute::Format::F32);
+
+    // validate that ACL can handle the given problem and inputs.
+    if (fuse_relu) {
+      arm_compute::Status relu_status =
```

Comment:
Why not use auto there?
```suggestion
      auto relu_status =
```

Reply:
Done [here](https://github.com/pytorch/pytorch/pull/148585/files#diff-db9f299d73216de73f5f8efb40022111c1e26eec780e49dbe70439cb7249013aR122)